### DISCUSSION TITLE
How to exclude shadcn from tsconfig in nextjs?

### DISCUSSION THREAD

**Discussion by gerkim62** [vercel/next.js#69279] [Help]
### Summary

I want typescript to not give me errors for files in the shadcn's components/ui folder files but `  "exclude": ["node_modules", "@/components/ui/**/*.tsx"],` is not fixing it.
![Screenshot 2024-08-24 205729](https://github.com/user-attachments/assets/3b63e087-a4ae-4f1b-8eb9-dadb27a19b9f)

### Additional information

```js
This is how my tsconfig.json looks like but am still getting errors from files located in components/ui/ folder. I am using the src directory together with app router. 

{
  "compilerOptions": {
    "lib": ["dom", "dom.iterable", "esnext", "WebWorker"],
    "target": "es6",
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules", "@/components/ui/**/*.tsx"],
  "extends": "@tsconfig/strictest/tsconfig.json"
}

notice i am using very strict tsconfig.json file.
```

**Comment by ZoranRavic** [comment]
You can't `exclude` files if they are imported by files in the `include` list.

You'll either have to fix those type errors or add `// @ts-nocheck` at the top of those files.

P.S. You can't write `@/folder/**/*.tsx` in `include` or `exclude`, you should write `folder/**/*.tsx` without the `@/`.
